client/my-sites/stats/pages/realtime/index.jsx

```diff
@@ -36,14 +39,45 @@ function StatsRealtime() {
 	// Need a period, a query, and a URL to use Top Posts.
 	// See getStatHref() example on Traffic page for URL.
 	const period = {};
+	const url = '#';
+
+	// TODO: Query is used inside useEffect block so we should memoize it.
+	// - BUT, we need to update the date when a new day starts so better to move it inside the block.
+
+	// TODO: Remove interval from query in favour of a real-time flag.
+
+	// TODO: Create a new query as part of the timed request.
+	// - This will make sure the date is correct when a new days starts.
+
+	// eslint-disable-next-line react-hooks/exhaustive-deps
 	const query = {
 		period: 'day',
 		date: momentSiteZone.format( 'YYYY-MM-DD' ),
 		max: 10,
 		summarize: 1,
 		interval: 10000, // Indicate the query should be periodically refreshed.
 	};
-	const url = '#';
+
+	useEffect( () => {
+		// TODO: This array determines which requests are fired.
+		// Currently firing two requests but only displaying top posts.
+		const statTypes = [ 'statsTopPosts', 'statsReferrers' ];
+
+		// Function to dispatch the request
+		const fetchStats = () => {
```

Comment:
If we want to query real-time data for modules at the page level, should the [old one in StatsModule](https://github.com/Automattic/wp-calypso/blob/e9f87c489716aea566963884a19cc19d2bce3160/client/my-sites/stats/stats-module/index.jsx#L83) be removed?

Reply:
Yes. I noted this in the Slack thread too but forgot to add a TODO in the code to call it out. Basically, I'll need a bit more time to clean up a few things after moving the requests to the page level.